src/Model/ModelManager.php

```diff
@@ -23,16 +25,43 @@
 use Sonata\DoctrineMongoDBAdminBundle\Datagrid\ProxyQuery;
 use Sonata\Exporter\Source\DoctrineODMQuerySourceIterator;
 use Symfony\Bridge\Doctrine\ManagerRegistry;
-use Symfony\Component\Form\Exception\PropertyAccessDeniedException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
 class ModelManager implements ModelManagerInterface
 {
     public const ID_SEPARATOR = '-';
+
+    /**
+     * @var ManagerRegistry
+     */
     protected $registry;
 
-    public function __construct(ManagerRegistry $registry)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    /**
+     * NEXT_MAJOR: Make $propertyAccessor mandatory.
+     */
+    public function __construct(ManagerRegistry $registry, ?PropertyAccessorInterface $propertyAccessor = null)
     {
         $this->registry = $registry;
+
+        // NEXT_MAJOR: Remove this block.
+        if (null === $propertyAccessor) {
+            @trigger_error(sprintf(
+                'Constructing "%s" without passing an instance of "%s" as second argument is deprecated since'
+                .' sonata-project/doctrine-mongodb-admin-bundle 3.x and will be mandatory in 4.0',
+                static::class,
+                PropertyAccessorInterface::class
```

Comment:
```suggestion
        if (!$propertyAccessor instanceof PropertyAccessorInterface) {
            @trigger_error(sprintf(
                'Not passing an object implementing "%s" as argument 2 for "%s()" is deprecated since'
                .' sonata-project/doctrine-mongodb-admin-bundle 3.x and will throw a %s error in 4.0.',
                PropertyAccessorInterface::class,
                __METHOD__,
                \TypeError::class                
```